### ISSUE TITLE
Updating editor interface for single field resets editor interfaces and help text for other fields

### ISSUE THREAD

**Issue by infrajatuki** [contentful/contentful-management.rb#155]
Hi,

I used the following code to change the editor interface of a single field in a content type. It appears that in the process some of the editor interfaces settings of other fields in the same content type have been changed/reset and the help texts have vanished.

```
client = Contentful::Management::Client.new(<TOKEN>)
space = client.spaces.find(<SPACE-ID>)
editor_interface = client.editor_interfaces.default(space_id, "content_type_id")

editor_interface.controls = [
  {
    fieldId: "fieldName",
    widgetId: "dropdown",
    settings: {
      helpText: "Here is some help text."
    }
  }
]

editor_interface.update(controls: editor_interface.controls)
```

**Comment by dlitvakb** [comment]
Hey @a-lehdet,

Sorry that happened, but this because the CMA API only allows PUT requests and not PATCH requests, so you need to specify the whole payload in order to update it without losing information.

What I recommend you to do is to update it by doing something like:

```ruby
client = Contentful::Management::Client.new(<TOKEN>)
space = client.spaces.find(<SPACE-ID>)
editor_interface = client.editor_interfaces.default(space_id, "content_type_id")

control = editor_interface.controls.detect { |ei| ei['fieldId'] == 'fieldName' }
control['settings'].update('helpText' => 'some help text') unless control.nil?

editor_interface.save
```

This will avoid losing data by removing existing configuration from the editor interface.

Hope this helps,

Cheers

**Comment by a-lehdet** [comment]
Thanks for the response and help @dlitvakb,

Now that you have explained it, it makes total sense given the structure of the payload. However, as this is a bit of "you just need to know it" issue, it might be worthwhile explaining this in the documentation more in-depth as it was not particularly intuitive nor obvious...not to me, at least. 😄

**Comment by a-lehdet** [comment]
Hi @dlitvakb,

As the example you provided no longer works with the current version, I could really use some help as the documentation is still scarce regarding editor interfaces and help texts. Basically what I'm trying to archive is loop through all the fields of a content type and set `widgetId` and `helpText` for each field.

If you could provide a code example how this can be done, I would appreciate it very much. :)

**Comment by dlitvakb** [comment]
Could you explain further what's not working? As there's no new published version since that comment has been made.

Cheers

**Comment by a-lehdet** [comment]
@dlitvakb I'm getting the following error.

```
Traceback (most recent call last):
1: from temp.rb:5:in `<main>'
/path/to/ruby/gems/contentful-management-2.1.1/lib/contentful/management/client.rb:229:in `editor_interfaces': wrong number of arguments (given 0, expected 3) (ArgumentError)
```

In addition, I'm a bit puzzled that the example connects to space rather than environment. I noticed that editor interfaces of a particular content type can be accessed like this...

`environment.content_types.find("content_type_id").editor_interface.default`

...but I wasn't able to figure out how to set values for `widgetId` and `helpText` of each field.

**Comment by dlitvakb** [comment]
Can you provide me the snippet that throws that error?

Cheers

**Comment by a-lehdet** [comment]
Here's the code. The purpose of the `editor_interface.controls.detect` loop is a bit unclear to me, but the content type in question `person` has a `fieldId` and `fieldName` that are called `name` and `Name`.

```
require "contentful/management"

client = Contentful::Management::Client.new("<TOKEN>")
space = client.spaces.find("<SPACE-ID>")
editor_interface = client.editor_interfaces.default(space.id, "person")

control = editor_interface.controls.detect do |c|
  c['name'] == 'Name'
end

control['settings'].update('helpText' => 'some help text') unless control.nil?

editor_interface.save
```

**Comment by a-lehdet** [comment]
@dlitvakb 

So far what I've got is this, basically just shooting in the dark...

person = @environment.content_types.find("person")
editor_interface = person.editor_interface.default

```
editor_interface.properties[:controls].each do |field|
  
  # Works, outputs the correct field value
  puts field["settings"]["helpText"] if defined? field["settings"]["helpText"]

  # Does nothing, but doesn't give any errors either
  field["settings"].update("helpText" => "testing") if defined? field["settings"]["helpText"]

end
```

Trying to save using `editor_interface.save` gives this return `unknown method` error. Using `person.save` instead doesn't return errors, but doesn't update the help text field either.

**Comment by dlitvakb** [comment]
Hey @a-lehdet,

I'll dig deeper into this as the snippet above (which I now updated to support environments) should work and will provide you an answer by around tomorrow this time.

Cheers

**Comment by dlitvakb** [comment]
Hey @a-lehdet,

This is indeed something that's currently wrong in the SDK, though it's not because it's a bug, it's just because it's implemented a bit differently.

I'm going to add a `#save` method to the SDK for editor interfaces to avoid creating a breaking change by rewriting `#update`.

I will create a PR and get it reviewed as soon as possible,

Cheers

**Comment by a-lehdet** [comment]
Hi @dlitvakb,

Thanks for investigating the issue. At the moment I've decided to proceed with the approach described in the first message of this thread (i.e. set the widgets and help texts in a single hash), but latter approach would definitely be more elegant, so adding the `save` method to editor interface is a welcome addition.

**Comment by dlitvakb** [comment]
This is the snippet with how the updated SDK works:

```ruby
require 'contentful/management'

client = Contentful::Management::Client.new(ENV['CF_TEST_CMA_TOKEN'])
space = client.spaces.find('facgnwwgj5fe')
environment = space.environments.find('master')
content_type = environment.content_types.find('cat')
editor_interface = content_type.editor_interface.default
control = editor_interface.controls.detect { |c| c['fieldId'] == 'name' }

if control['settings'].nil?
  control['settings'] = {'helpText' => 'some help text'}
else
  control['settings'].update('helpText' => 'some help text')
end

editor_interface.save
```

Creating the PR right now,

Cheers

**Comment by dlitvakb** [comment]
This has just been released as version 2.2.0

Cheers
